src/transactions/feed/queryHelper.ts

```diff
@@ -57,6 +58,9 @@ const deduplicateTransactions = (
   return transactionsWithoutDuplicatedHash
 }
 
+// A map of transactions by tx hash
+const transactionMap = new Map<string, TokenTransaction>()
```

Comment:
okay cool that does simplify the implementation. @jeanregisser we had talked about comparing against all stored transactions in case we cannot rely on the order of events from blockscout (like in notification service we have that elaborate way to index events because blockscout doesn't always return the most recent events first, not sure if that applies to the blockscout service that blockchain-api uses though) - we're sure that this case doesn't apply here? if for whatever reason we receive [tx1, tx3, tx4] and then [tx1, tx2, tx3, tx4] from blockchain-api and we only check against the last tx hash, then the feed wouldn't be updated. :/

Reply:
ah the eventual consistency problem, yes you're right!
Sorry @MuckT.

Reply:
@kathaypacific @jeanregisser I went ahead and used a redux selector in 29968044f41f316fd5705253222570f5552f6dc1. Hopefully this resolves the issue!